src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -359,6 +361,10 @@ private class TaskLauncherActor(
       sender ! MatchedInstanceOps(offer.getId)
 
     case ActorOfferMatcher.MatchOffer(deadline, offer) =>
+      val active = instanceMap.values.filter(i => i.isActive).map(i => i.instanceId).mkString(",")
+      val reserved = instanceMap.values.filter(i => i.isReserved).map(i => i.instanceId).mkString(",")
+      val lost = instanceMap.values.filter(i => i.state.condition.isLost).map(i => i.instanceId).mkString(",")
+      log.debug(s"active ins [$active] reserved [$reserved] lost[$lost] reserveInFlight ${reserveInFlight.size}")
       val reachableInstances = instanceMap.filterNotAs{ case (_, instance) => instance.state.condition.isLost }
       val matchRequest = InstanceOpFactory.Request(runSpec, offer, reachableInstances, instancesToLaunch)
```

Comment:
Please correct me if I am wrong but we should be passing in `reserveInFlight` (or a copy) to `InstanceOpFactory.Request()`, no?